defusedcsv/csv.py

```diff
@@ -12,37 +12,40 @@
            "unregister_dialect", "DictReader", "DictWriter", "unix_dialect"]
 
 
-def escape(payload):
+def escape(payload, assert_injection=False):
     if payload is None:
         return ''
 
     payload = str(payload)
     if payload and payload[0] in ('@', '+', '-', '=', '|', '%') and not re.match("^-?[0-9,\\.]+$", payload):
+        if assert_injection is True:
+            raise Error('The payload which shall be written is dangerous (CSV injection).')
         payload = payload.replace("|", "\\|")
         payload = "'" + payload
     return payload
 
 
 class ProxyWriter:
-    def __init__(self, writer):
+    def __init__(self, writer, assert_injection):
```

Comment:
Sorry I didn't reply to the issue any more yesterday, got busy with other stuff. I don't think ``assert_injection`` is a perfect name, it kinda sounds like the opposite: "assert that there's an injection", I think that's not what we want. Maybe something like ``fail_when_unsafe=False`` or, if we turn the logic around, a parameter ``escape_unsafe=True`` that needs to be set to ``False`` for the exception.

Reply:
I'd go with the logic as is and `fail_when_unsafe` instead of inverting the logic and  `escape_unsafe`. But it's your project so just let me know what you prefer to get the functionality in :wink: